From 35985c82b0d5303af4131fed1d0faa47241dd169 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Tue, 24 Oct 2017 13:09:27 +0300
Subject: [PATCH 2681/5242] MLK-13946-8: ASoC: fsl_sai: use
 min(channels,slots) for xMR setting

commit  66549182abd27188242cacbedd38e54bbca1326d from
https://source.codeaurora.org/external/imx/linux-imx.git

xMR setting must be set as min(channels,slots) since
both "channels < slots" and "channels > slots" scenarios
are possible.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_sai.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index abe0854..dbb584e 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -569,8 +569,8 @@ static int fsl_sai_hw_params(struct snd_pcm_substream *substream,
 	regmap_update_bits(sai->regmap, FSL_SAI_xCR5(tx, offset),
 			   FSL_SAI_CR5_WNW_MASK | FSL_SAI_CR5_W0W_MASK |
 			   FSL_SAI_CR5_FBT_MASK, val_cr5);
-	regmap_write(sai->regmap, FSL_SAI_xMR(tx), ~0UL - ((1 << slots) - 1));
-
+	regmap_write(sai->regmap, FSL_SAI_xMR(tx),
+			~0UL - ((1 << min(channels, slots)) - 1));
 	return 0;
 }
 
-- 
1.7.9.5

